src/tsid.cpp

```diff
@@ -139,13 +133,17 @@ void setup() {
 		EEPROM.put(EEPROM_ADDR_VERSION, EEPROM_FORMAT_VERSION_V2);
 	}
 
+	// are we holding reset (to reset globals)?
+	mux(7);
+	bool initialize_globals = ((PINA & _BV(4)) == 0);
+
 	// Update all global settings from EEPROM memory
 	for (int i = 0; i < (int)(sizeof(globalSettings) / sizeof(globalSetting)); i++) {
 		const globalSetting* setting = &(globalSettings[i]);
 		byte value = EEPROM.read(setting->eepromAddress);
 
 		// Validate range and set default if needed
-		if ((value < setting->minValue) || (value > setting->maxValue)) {
+		if ((value < setting->minValue) || (value > setting->maxValue) || initialize_globals) {
```

Comment:
@twisted-electrons There are some more details here :) An EEPROM.update() is needed to be added as well since it is currently only changed in-memory. Purpose from start with the "validate" was basically to fix errors that for some reason would be stored in the EEPROM, but now with initialize_globals added it also requires writing it back.

Reply:
@thomasj  Indeed!! Thanks for spotting that too!!